### PR TITLE
Add pre-commit hooks for linting/testing

### PR DESCRIPTION
**Pull request by jestapinski**
Now that `npm test` works again, this PR enables linting and testing on commits (similar to with university) using husky/lint-staged.

After enabling this, there were some lint errors to handle, so these were taken care of as well.

### CODE REVIEW

File: preview-start.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
+/* eslint-disable import/no-unresolved */
```
Comment (jestapinski) [inline preview-start.js]: The `guides` and `guides-index` are not in this repo, so perhaps we should omit them here. I am not too familiar with the preview work, however.

Comment (greggb) [reply preview-start.js]: I'm hazy on this as well. We'll probably need to use our own templates, but there is still work to be done on the docs platform side as well

File: src/components/dev-hub/controlled-tooltip.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-unused-vars */
```
Comment (greggb) [inline src/components/dev-hub/controlled-tooltip.js]: What's the unused var in this file?

File: tests/unit/use-event-data.test.js [hunk 3]
```diff
@@ -41,6 +41,7 @@ const timezonedEvents = [
 ];
 it('should remove events that have past', () => {
     const filtered = removePastEvents(events);
+    // eslint-disable-next-line no-unused-vars
```
Comment (greggb) [inline tests/unit/use-event-data.test.js]: Should we just not assign a variable below?

File: src/components/dev-hub/controlled-tooltip.js (hunk 2 above)
Comment (jestapinski) [reply src/components/dev-hub/controlled-tooltip.js]: There were a few props in the component not being used. I saw a lengthy comment about how this component was used and decided it was safer just to add the comment. We can remove them I have no issue with that.

File: tests/unit/use-event-data.test.js (hunk 3 above)
Comment (jestapinski) [reply tests/unit/use-event-data.test.js]: Fair point

File: src/components/dev-hub/controlled-tooltip.js (hunk 2 above)
Comment (greggb) [reply src/components/dev-hub/controlled-tooltip.js]: I think they may have a different lint setup where you can't pull out spread items without the linter warning - I'm fine either way

Comment (jestapinski) [reply src/components/dev-hub/controlled-tooltip.js]: I just went ahead and pulled them out

**Comment by jestapinski** [comment]
Thanks!

**Pull request merged**